src/store/application/ui/average-color/slice.ts

```diff
@@ -0,0 +1,37 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import Color from 'models/common/Color'
+import { CID } from 'models/common/Identifiers'
+import { AppState } from 'store/types'
+import { Nullable } from 'utils/typeUtils'
+
+const initialState: { colors: { [multihash: string]: Color } } = {
+  colors: {}
+}
+
+/**
+ * This slice tracks computed average colors for a given track art CID.
+ * Colors is a map of art cid -> Color
+ */
+const slice = createSlice({
+  name: 'application/ui/averageColor',
+  initialState,
+  reducers: {
+    setColor: (
+      state,
+      action: PayloadAction<{ multihash: string; color: Color }>
+    ) => {
+      const { multihash, color } = action.payload
+      state.colors[multihash] = color
+    }
+  }
+})
+
+export const { setColor } = slice.actions
+
+export const getAverageColor = (
+  state: AppState,
+  { multihash }: { multihash: Nullable<CID> }
```

Comment:
maybe nitty, but might be nice to export a second selector here that takes state and track and does the logic you did in the mapstateto's

Reply:
yeah dis is a good idea, will do